src/java/org/apache/nutch/scoring/ScoringFilter.java

```diff
@@ -179,6 +179,20 @@ public void updateDbScore(Text url, CrawlDatum old, CrawlDatum datum,
       List<CrawlDatum> inlinked) throws ScoringFilterException;
 
   /**
+   * This method may change the score or status of CrawlDatum during CrawlDb
+   * update, when the URL is neither fetched nor has any inlinks.
+   *
+   * @param url
+   *          URL of the page
+   * @param datum
+   *          CrawlDatum for page
+   * @throws ScoringFilterException
```

Comment:
Ok, thanks. However, the current state would require more fixes. `@throws ScoringFilterException` hasn't a description anywhere, it's even not consistently "declared" in the Javadoc. Let's keep this for a later improvement.